packages/client/src/lib/interface.ts

```diff
@@ -26,7 +50,13 @@ export interface PublicService {
  */
 export type CIDString = Tagged<string, CID>
 
-export interface API {
+/**
+ * The public interface for the static methods of the NFT.Storage client API. See {@link NFTStorage} for the
+ * implementation.
+ */
+export interface StaticAPI {
+  new (options: { token: string; endpoint?: URL }): InstanceAPI
```

Comment:
```suggestion
```

Can we not have this ? Idea behind API (now StaticAPI) is that other component is that:

1. Alternative compatible API can be implemented by other party. Requiring that it also be class is unnecessary complication.
2. Component can request API implementation to be passed to it so it can exercise it. Requirement of it being a class that constructs non static instance is unnecessary burden to satisfy. If components withes non static API it should probably just require that as an argument instead no matter how it was instantiated.

Reply:
If you want ensure our class is compatible in that case I would just break out this into separate `InstanceFactory` interface and do:

```js
/**
 * Just to verify API compatibility.
 * @type {API.StaticAPI & API.InstanceFactory}
 */
const api = NFTStorage
```